broadcaster/_backends/redis.py

```diff
@@ -1,34 +1,50 @@
-import asyncio_redis
+import aioredis
+import asyncio
 import typing
-from urllib.parse import urlparse
 from .base import BroadcastBackend
 from .._base import Event
 
 
 class RedisBackend(BroadcastBackend):
     def __init__(self, url: str):
-        parsed_url = urlparse(url)
-        self._host = parsed_url.hostname or "localhost"
-        self._port = parsed_url.port or 6379
+        self.conn_url = url
+
+        self._pub_conn: typing.Optional[aioredis.Redis] = None
+        self._sub_conn: typing.Optional[aioredis.Redis] = None
+
+        self._msg_queue: typing.Optional[asyncio.Queue] = None
+        self._tasks: typing.List[asyncio.Task] = []
 
     async def connect(self) -> None:
-        self._pub_conn = await asyncio_redis.Connection.create(self._host, self._port)
-        self._sub_conn = await asyncio_redis.Connection.create(self._host, self._port)
-        self._subscriber = await self._sub_conn.start_subscribe()
+        self._pub_conn = await aioredis.create_redis(self.conn_url)
+        self._sub_conn = await aioredis.create_redis(self.conn_url)
+        self._msg_queue = asyncio.Queue()  # must be created here, to get proper event loop
 
     async def disconnect(self) -> None:
         self._pub_conn.close()
         self._sub_conn.close()
 
+        self._pub_conn = None
+        self._sub_conn = None
+        self._msg_queue = None
+
     async def subscribe(self, channel: str) -> None:
-        await self._subscriber.subscribe([channel])
+        channels = await self._sub_conn.subscribe(channel)
+        self._tasks.append(asyncio.create_task(self.reader(channels[0]), name=f"{channel} reader"))
```

Comment:
How will this scale when we have dozens of tasks running on all active notebooks?

Reply:
I think really what I'm asking here is if there's a way for us to have a single task get updated with all the channels we're trying to watch, but I don't actually know if having many tasks for this is a significant problem?

Reply:
Good call, aioredis has a `Receiver` class that we can use to have a single listener task

Reply:
I thought we could do:

```python
async def next_published(self) -> Event:
	channel, msg = await self._mpsc.get()
	return Event(...)
```

but `.get()` will return `None` when the queue is closed. I couldn't find a way around this, so I think just having the single reader task will be fine.